File: Ruby/Irrgarten/controller.rb
module Control

  class Controller
    def initialize(game,view)
      @game = game
      @view = view
    end

    def play
      end_of_game = false
      while (!end_of_game)
        @view.show_game(@game.getGameState())
        direction = @view.next_move
        puts("0_111111111111111111111111111")
        end_of_game = @game.nextStep(direction)
        puts("1_111111111111111111111111111")
      end
      @view.show_game(@game.gameState)
    end
  end # class   
end # module        

File: Java/irrgarten/src/irrgarten/controller/Controller.java
package irrgarten.controller;

import irrgarten.modelo.Directions;
import irrgarten.modelo.Directions;
import irrgarten.modelo.Game;
import irrgarten.modelo.Game;
import irrgarten.UI.TextUI;
import irrgarten.modelo.*;
import irrgarten.UI.GraphicalUI;

public class Controller {
    
    private Game game;
    private GraphicalUI view;
    
    public Controller(Game game, GraphicalUI view) {
        this.game = game;
        this.view = view;
    }
    
    public void play() {
        boolean endOfGame = false;
        while (!endOfGame) {
            view.showGame(game.getGameState()); 
            Directions direction = view.nextMove(); 
            endOfGame = game.nextStep(direction);
        }
      view.showGame(game.getGameState());        
    }
    
}
